src/sincosint.jl

```diff
@@ -0,0 +1,258 @@
+const pidiv2 = π/2
+
+import Base.Math: @horner
+
+# Compute the sine integral: ∫_0^x sin(t)/t dt,
+# and the cosine integral: γ + log x + ∫_0^x (cos(t)-1)/t dt,
+# using the rational approximants tabulated in:
+#     A.J. MacLeod, "Rational approximations, software and test methods for
+#     sine and cosine integrals," Numer. Algor. 12, pp. 259--272 (1996).
+#         http://dx.doi.org/10.1007/BF02142806
+#         https://link.springer.com/article/10.1007/BF02142806
+#
+# Note: the second zero of Ci(x) has a typo that is fixed:
+#
+# r1 = 3.38418 04228 51186 42639 78511 46402 in the article, but is in fact:
+#
+# r1 = 3.38418 04225 51186 42639 78511 46402.
+#
+
+function sinint(x::Float64)
+    t = x*x
+    if t ≤ 36.0
+        return x * @horner(t, 1.00000000000000000000E0,
+                             -0.44663998931312457298E-1,
+                              0.11209146443112369449E-2,
+                             -0.13276124407928422367E-4,
+                              0.85118014179823463879E-7,
+                             -0.29989314303147656479E-9,
+                              0.55401971660186204711E-12,
+                             -0.42406353433133212926E-15) /
+                   @horner(t, 1.00000000000000000000E0,
+                              0.10891556624243098264E-1,
+                              0.59334456769186835896E-4,
+                              0.21231112954641805908E-6,
+                              0.54747121846510390750E-9,
+                              0.10378561511331814674E-11,
+                              0.13754880327250272679E-14,
+                              0.10223981202236205703E-17)
+    elseif t ≤ 144.0
+        invt = inv(t)
```

Comment:
I guess, in this particular instance, it's just a question of style because `t::Float64` and all three cases `1/t`, `1.0/t`, and `inv(t)` get lowered to the same code.

I changed it to `1.0/t` because perhaps in the future not all languages will support `inv` on a floating-point number (and someone may base, however abstractly to avoid licensing issues, another implementation off this one). If the function were more generic, then I probably would've left it as `inv` because of non-trivial improvements for different types.